server/autotest_server/utils.py

```diff
@@ -45,31 +43,54 @@ def _rlimit_str2int(rlimit_string):
     return getattr(resource, f"RLIMIT_{rlimit_string.upper()}")
 
 
-def set_rlimits_before_test() -> None:
+def validate_rlimit(config_soft: int, config_hard: int, curr_soft: int, curr_hard: int) -> tuple[int, int]:
+    """Validates and adjusts resource limits based on configured and current values.
+
+    Returns a tuple containing the validated (soft, hard) limit values. This implementation treats the current soft
+    limit as an upper bound on the config soft limit and will clamp it.
     """
-    Sets rlimit settings specified in config file
+    # account for the fact that resource.RLIM_INFINITY == -1
+    soft, hard = min(curr_soft, config_soft), min(curr_hard, config_hard)
+    if soft < 0:
+        soft = max(curr_soft, config_soft)
+    if hard < 0:
+        hard = max(curr_hard, config_hard)
+    # make sure the soft limit doesn't exceed the hard limit, but keep in mind that -1 is resource.RLIM_INFINITY
+    if hard != -1:
+        soft = min(hard, soft)
+
+    return soft, hard
+
+
+def get_resource_settings(config: _Config) -> list[tuple[int, tuple[int, int]]]:
+    """Returns rlimit settings specified in config file.
+
     This function ensures that for specific limits (defined in RLIMIT_ADJUSTMENTS),
     there are at least n=RLIMIT_ADJUSTMENTS[limit] resources available for cleanup
```

Comment:
I think `RLIMIT_ADJUSTMENTS` no longer applies, this documentation should be updated.